src/interface/contract.rs

```diff
@@ -121,6 +122,58 @@ pub trait StateChange: Clone + Eq + StrictDumb + StrictEncode + StrictDecode {
     fn merge_received(&mut self, state: Self::State);
 }
 
+#[derive(Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Display)]
+#[derive(StrictType, StrictDumb, StrictEncode, StrictDecode)]
+#[strict_type(lib = LIB_NAME_RGB_STD, tags = custom)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate", rename_all = "camelCase")
+)]
+pub enum OwnedFractionChange {
+    #[display("-{0}")]
+    #[strict_type(tag = 0xFF)]
+    Dec(OwnedFraction),
+
+    #[display("0")]
+    #[strict_type(tag = 0, dumb)]
+    Zero,
+
+    #[display("+{0}")]
+    #[strict_type(tag = 0x01)]
+    Inc(OwnedFraction),
+}
+
+impl StateChange for OwnedFractionChange {
+    type State = OwnedFraction;
+
+    fn from_spent(state: Self::State) -> Self { OwnedFractionChange::Dec(state) }
+
+    fn from_received(state: Self::State) -> Self { OwnedFractionChange::Inc(state) }
+
+    fn merge_spent(&mut self, sub: Self::State) {
+        *self = match self {
+            OwnedFractionChange::Dec(neg) => OwnedFractionChange::Dec(*neg + sub),
+            OwnedFractionChange::Zero => OwnedFractionChange::Dec(sub),
+            OwnedFractionChange::Inc(pos) if *pos > sub => OwnedFractionChange::Inc(*pos - sub),
+            OwnedFractionChange::Inc(pos) if *pos == sub => OwnedFractionChange::Zero,
+            OwnedFractionChange::Inc(pos) if *pos < sub => OwnedFractionChange::Dec(sub - *pos),
+            OwnedFractionChange::Inc(_) => unreachable!(),
+        };
```

Comment:
with `Ord` you can get rid of `unreachable` and instead enjoy compiler-provided exhaustive security for matches:

```suggestion
        *self = match self {
            OwnedFractionChange::Dec(neg) => OwnedFractionChange::Dec(*neg + sub),
            OwnedFractionChange::Zero => OwnedFractionChange::Dec(sub),
            OwnedFractionChange::Inc(pos) => match sub.cmp(pos) {
              Ordering::Less => OwnedFractionChange::Inc(*pos - sub),
              Ordering::Equal => OwnedFractionChange::Zero,
              Ordering::Greater => OwnedFractionChange::Dec(sub - *pos),
            },
        };
```